app/src/main/java/de/htwdd/htwdresden/ManagementFragment.java

```diff
@@ -66,6 +89,131 @@ public void onClick(View view) {
+            } //else System.out.println("TAKEN FROM SCHARED_PREFS");
+        }
+
+        swipeRefrSemPlan.setOnRefreshListener(
+                new SwipeRefreshLayout.OnRefreshListener() {
+                    @Override
+                    public void onRefresh() {
+                        //Log.i("REFRESH CALLED", "onRefresh called from SwipeRefreshLayout");
+                        writeToSharedPrefAndSendReq(jsonArrayListener, errorListener, sharedPref);
+                        updateSemPlanView(view, sharedPref);
+                        swipeRefrSemPlan.setRefreshing(false);
+                    }
+                }
+        );
+
+        updateSemPlanView(view, sharedPref);
+
         return view;
     }
+
+    private void updateSemPlanView(View view, SharedPreferences sharedPref) {
+        try {
+            String sJSON = sharedPref.getString(SHARED_PREFS_SEMESTER_PLAN_JSON, "LEER");
+            //Log.i("UPDATING VIEW", sJSON);
+            JSONObject jsonObject = new JSONObject(sJSON);
+            SemesterPlan semesterPlan = new SemesterPlan(jsonObject);
+            setSemesterPlanView(semesterPlan, view);
+        } catch (JSONException e) {
+            System.out.println("StundenPlanJSON not ready yet");
+        }
+    }
+
+    private void setSemesterPlanView(SemesterPlan s, View view) {
+        final TextView semesterplanBezeichnung = (TextView) view.findViewById(R.id.semesterplan_Bezeichnung);
+        final TextView semesterplanLecturePeriod = (TextView) view.findViewById(R.id.semesterplan_lecturePeriod);
+        final TextView semesterplanFreieTage = (TextView) view.findViewById(R.id.semesterplan_freieTage);
+        final TextView semesterplanFreieTageNamen = (TextView) view.findViewById(R.id.semesterplan_freieTageNamen);
+        final TextView semesterplanPruefPeriod = (TextView) view.findViewById(R.id.semesterplan_pruefPeriod);
+        final TextView semesterplanRegistration = (TextView) view.findViewById(R.id.semesterplan_reregistration);
+
+        semesterplanBezeichnung.setText(s.getBezeichnung());
+        semesterplanLecturePeriod.setText(s.getLecturePeriod().toString());
+        semesterplanFreieTageNamen.setText("");
+        semesterplanFreieTage.setText("");
+        for (FreeDay f : s.getFreeDays()) {
+            semesterplanFreieTageNamen.append(f.getNAME());
+            semesterplanFreieTage.append(f.toString());
+        }
+        semesterplanPruefPeriod.setText(s.getExamsPeriod().toString());
+        semesterplanRegistration.setText(s.getReregistration().toString());
+    }
+
+    private void writeToSharedPrefAndSendReq(Response.Listener<JSONArray> jsonArrayListener, Response.ErrorListener errorListener,SharedPreferences sharedPref) {
+        long currentTime = System.currentTimeMillis();
+        SharedPreferences.Editor editor = sharedPref.edit();
+        editor.putLong(SHARED_PREFS_CATCH_DATE, currentTime);
+        editor.apply();
```

Comment:
Ihr speichert bereits den Zeitpunkt der letzten Aktualisierung, obwohl noch nicht mal der Request abgeschickt wurde! Im schlimmsten Fall landet der Resonse im errorListener, dann erfolgt die nächste trotzdem erst in 3 Wochen.

Bitte in den jsonArrayListener verschieben, nachdem alle Daten geparst und gespeichert sind, kann dieses Flag gesetzt werden.